sdv/multi_table/hma.py

```diff
@@ -525,12 +538,17 @@ def _extract_parameters(self, parent_row, table_name, foreign_key):
     def _recreate_child_synthesizer(self, child_name, parent_name, parent_row):
         # A child table is created based on only one foreign key.
         foreign_key = self.metadata._get_foreign_keys(parent_name, child_name)[0]
-        parameters = self._extract_parameters(parent_row, child_name, foreign_key)
-        default_parameters = getattr(self, '_default_parameters', {}).get(child_name, {})
 
-        table_meta = self.metadata.tables[child_name]
-        synthesizer = self._synthesizer(table_meta, **self._table_parameters[child_name])
-        synthesizer._set_parameters(parameters, default_parameters)
+        if parent_row is not None:
+            parameters = self._extract_parameters(parent_row, child_name, foreign_key)
+            default_parameters = getattr(self, '_default_parameters', {}).get(child_name, {})
+
+            table_meta = self.metadata.tables[child_name]
+            synthesizer = self._synthesizer(table_meta, **self._table_parameters[child_name])
+            synthesizer._set_parameters(parameters, default_parameters)
+        else:
+            synthesizer = self._null_child_synthesizers[f'__{child_name}__{foreign_key}']
```

Comment:
I think this will fail backwards compatibility. If we load a synthesizer that was saved before the release with these changes and then try to sample, this attribute won't exist and will crash

Reply:
I think this particular code path won't be reached by old synthesizers since for `parent_row` to be `None`, `HMASynthesizer.null_foreign_key_percentages` has to be non-zero for the table.  Old synthesizers won't have the `null_foreign_key_percentages` attribute and default to `0` as the percentage.